src/envs/behavior_options.py

```diff
@@ -0,0 +1,976 @@
+    old_pos = np.array([old_xytheta[0], old_xytheta[1], robot_z])
+    old_orn_quat = p.getQuaternionFromEuler(
+        np.array(
+            [original_orientation[0], original_orientation[1],
+             old_xytheta[2]]))
+    new_pos = np.array([new_xytheta[0], new_xytheta[1], robot_z])
+    new_orn_quat = p.getQuaternionFromEuler(
+        np.array(
+            [original_orientation[0], original_orientation[1],
+             new_xytheta[2]]))
+
+    # Then, simply get the delta position and orientation by multiplying the
+    # inverse of the old pose by the new pose
+    inverted_old_pos, inverted_old_orn_quat = p.invertTransform(
+        old_pos, old_orn_quat)
+    delta_pos, delta_orn_quat = p.multiplyTransforms(inverted_old_pos,
+                                                     inverted_old_orn_quat,
+                                                     new_pos, new_orn_quat)
+
+    # Finally, convert the orientation back to euler angles from a quaternion
+    delta_orn = p.getEulerFromQuaternion(delta_orn_quat)
+
+    ret_action[0:3] = np.array([delta_pos[0], delta_pos[1], delta_orn[2]])
+
+    return ret_action
+
+
+# NOTE: we need a type:ignore because we can't specify the type of objects
+# because it would be dependent on an optional igibson import which won't
+# get imported on CI
+def navigate_to_param_sampler(  # type: ignore
+        rng: Generator, objects) -> Array:
+    """Sampler for navigateTo option."""
+    assert len(objects) in [2, 3]
+    # The navigation nsrts are designed such that this is true (the target
+    # obj is always last in the params list).
+    obj_to_sample_near = objects[-1]
+    closeness_limit = max(
+        [
+            1.5,
+            np.linalg.norm(np.array(\
+                obj_to_sample_near.bounding_box[:2])) + 0.5,  # type: ignore
+        ]
+    )
+    distance = (closeness_limit - 0.01) * rng.random() + 0.03  # type: ignore
+    yaw = rng.random() * (2 * np.pi) - np.pi
+    x = distance * np.cos(yaw)
+    y = distance * np.sin(yaw)
+
+    # The below while loop avoids sampling values that are inside
+    # the bounding box of the object and therefore will
+    # certainly be in collision with the object if the robot
+    # tries to move there.
+    while (abs(x) <= obj_to_sample_near.bounding_box[0]
+           and abs(y) <= obj_to_sample_near.bounding_box[1]):
+        distance = (
+            closeness_limit -  # type: ignore
+            0.01) * rng.random() + 0.03
+        yaw = rng.random() * (2 * np.pi) - np.pi
+        x = distance * np.cos(yaw)
+        y = distance * np.sin(yaw)
+
+    return np.array([x, y])
+
+
+# NOTE: we need a type:ignore because we can't specify the type of env because
+# it would be a igibson.envs.behavior_env.BehaviorEnv, which is an optional
+# import and won't get imported on CI
+def navigate_to_obj_pos(  # type: ignore
+        env,
+        obj,
+        pos_offset: Array,
+        rng: Generator = np.random.default_rng(23),
```

Comment:
generally we want to avoid mutable default arguments. for example, if there are multiple calls to `navigate_to_obj_pos`, they would share a single rng. could we make this kwarg optional and initialize it within the function if it's None?